compiler/rustc_codegen_llvm/src/back/archive.rs

```diff
@@ -175,6 +189,70 @@ impl<'a> ArchiveBuilder<'a> for LlvmArchiveBuilder<'a> {
             self.config.sess.fatal(&format!("failed to build archive: {}", e));
         }
     }
+
+    fn inject_dll_import_lib(
+        &mut self,
+        lib_name: Symbol,
+        dll_imports: &[DllImport],
+        tmpdir: &MaybeTempDir,
+    ) {
+        let output_path = {
+            let mut output_path: PathBuf = tmpdir.as_ref().to_path_buf();
+            output_path.push(format!("{}_imports", lib_name));
+            output_path.with_extension("lib")
+        };
+
+        // we've checked for \0 characters in the library name already
+        let dll_name_z = CString::new(format!("{}.dll", lib_name)).unwrap();
```

Comment:
This should respect the `verbatim` link modifier. By default it's false and in that case we should append the `.dll` suffix but if it is true then it should be passed as-is.

Reply:
This raises an interesting question: what should rustc do if there are multiple extern blocks for the same library with different verbatim modifiers?  The current implementation combines all extern blocks for the same library (identified by name, since that's all we have to go on) into a single import library.

As a concrete example, should we combine the symbols from the following two extern blocks?
```
#[link(name = "foo", kind = "raw-dylib")]
extern { ... }

#[link(name = "foo.dll", kind = "raw-dylib"), modifiers = "+verbatim")]
extern { ... }
```

Reply:
In that case I'd say those 2 blocks should be combined. That is, still key it by name but append the `.dll` suffix in the non-verbatim cases, i.e. moving the append logic earlier to `collate_raw_dylibs`